.github/workflows/test-and-publish-nightly.yml

```diff
@@ -65,65 +136,58 @@ jobs:
           architecture: ${{ matrix.architecture }}
       - name: Checkout sources
         uses: actions/checkout@v4
-      - name: Build and test
-        id: build
+      - name: Set artifact name
+        id: set_artifact_name
+        run: echo "artifact_name=async-profiler-${GITHUB_SHA:0:7}-${{ matrix.asprof-binaries-job || matrix.runson.display }}" >> $GITHUB_OUTPUT
+        shell: bash
+        env:
+          GITHUB_SHA: ${{ github.sha }}
+      - name: Download async-profiler release artifact
+        uses: actions/download-artifact@v4
+        with:
+          name: ${{ steps.set_artifact_name.outputs.artifact_name }}
+          path: async_profiler_release
+      - name: Download async-profiler JAR artifacts
+        uses: actions/download-artifact@v4
+        with:
+          name: async-profiler-jars
+          path: jar_artifacts
+      - name: Extract async-profiler artifact
+        id: extract_artifact
         run: |
-          HASH=$(echo ${{ github.sha }} | cut -c-7)
+          release_archive=$(basename $(find async_profiler_release -type f -iname "async-profiler-*" ))
           case "${{ matrix.runson.name }}" in
             macos*)
-              make COMMIT_TAG=$HASH FAT_BINARY=true release test -j
+              unzip async_profiler_release/$release_archive
             ;;
             *)
-              make COMMIT_TAG=$HASH CC=/usr/local/musl/bin/musl-gcc release test -j
+              tar xvf async_profiler_release/$release_archive
             ;;
           esac
-          echo "archive=$(basename $(find . -type f -iname "async-profiler-*" ))" >> $GITHUB_OUTPUT
-      - name: Coverage
-        id: coverage
+          echo "jars_directory=jar_artifacts" >> $GITHUB_OUTPUT
+          echo "release_directory=$(basename $(find . -type d -iname "async-profiler-*" ))" >> $GITHUB_OUTPUT
+      - name: Run integration tests
```

Comment:
We could move this to the `Makefile`, as `make integ ASPROF_BUILD=<asprof build folder>`, to enable "integration testing" local async-profiler binaries from other folders.

Reply:
https://github.com/async-profiler/async-profiler/pull/1238

Reply:
Can't we do both, add a makefile target that does not do any copy but changes paths to read from `ASPROF_BUILD` and call this from GHA?

The use case I have in mind is running same tests in different locally checked-out async-profiler folders.

Reply:
Does it need to be another target? Maybe it's enough to parametrize `build/`

Reply:
When `ASPROF_BUILD` is passed, we do not want to build makefile dependencies of `test-java`, otherwise they should be built as is today.

Reply:
Let's consider this separately and not block the PR.